lib/services/create_basket_item.rb

```diff
@@ -1,15 +1,23 @@
 module Shop
   class CreateBasketItem
+    attr_reader :product_id
+    attr_accessor :quantity
+
     def initialize(params)
       @product_id = params.fetch("product_id").to_i
       @quantity = params.fetch("quantity").to_i
     end
 
     def call
-      BASKET_ITEMS << BasketItem.new(
-        product_id: @product_id,
-        quantity: @quantity
-      )
+      item_in_basket = BASKET_ITEMS.find { |item| item.product_id == @product_id }
```

Comment:
Refer to product_id using method instead of instance variable `@product_id` (you've created  attr_reader)